malcolm/modules/ADPandABlocks/parts/kinematicssavupart.py

```diff
@@ -0,0 +1,353 @@
+        self.use_min_max = True
+        self.savu_file = None
+        self.layout_table = None
+        self.pos_table = None
+        self.cs_port = cs_port
+        self.cs_mri_suffix = cs_mri_suffix
+        self.shape = None
+        self.pmac_mri = None
+        self.panda_mri = None
+        self.axis_numbers = {}
+        self.generator = None
+
+    def setup(self, registrar):
+        # type: (PartRegistrar) -> None
+        super(KinematicsSavuPart, self).setup(registrar)
+        print("In kinsav setup")
+        # Tell the controller to expose some extra configure parameters
+        registrar.report(scanning.hooks.ConfigureHook.create_info(
+            self.configure))
+        # Hooks
+        registrar.hook(scanning.hooks.ConfigureHook, self.configure)
+        registrar.hook(scanning.hooks.PostConfigureHook, self.post_configure)
+
+    # Allow CamelCase as these parameters will be serialized
+    # noinspection PyPep8Naming
+    @add_call_types
+    def configure(self,
+                  context,  # type: scanning.hooks.AContext
+                  fileDir,  # type: scanning.hooks.AFileDir
+                  generator,  # type: scanning.hooks.AGenerator
+                  axesToMove,  # type: scanning.hooks.AAxesToMove
+                  fileTemplate="%s.nxs",  # type: scanning.hooks.AFileTemplate
+                  ):
+        # type: (...) -> None
+
+        self.p_vars = []
+        self.use_min_max = True
+        self.savu_variables = {}
+        self.savu_code_lines = []
+        self.shape = generator.shape
+        self.q_value_mapping = {}
+        self.generator = generator
+
+        # On initial configure, expect to get the demanded number of frames
+        child = context.block_view(self.mri)
+        self.pmac_mri = child.pmac.value
+        self.panda_mri = child.panda.value
+
+        # Derive file path from template
+        baseTemplate = os.path.splitext(fileTemplate)[0]
+        # Create the various nexus files to pass to Savu and expected output
+        fileName = baseTemplate.replace('%s', "savu") + ".nxs"
+        vds_fileName = baseTemplate.replace('%s', "kinematics-vds") + ".nxs"
+        savu_pl_fileName = baseTemplate.replace('%s', "savu_pl") + ".nxs"
+        savu_fileName = baseTemplate.replace('%s', "savu_processed") + ".nxs"
+
+        # This is path to the file to pass to Savu
+        self.nxs_full_filename = os.path.join(fileDir, fileName)
+
+        # This is path to the process list file to pass to Savu
+        self.savu_pl_filename = os.path.join(fileDir, savu_pl_fileName)
+
+        # This is the path to the VDS file which links to the processed Savu
+        # file with the output datasets
+        self.vds_full_filename = os.path.join(fileDir, vds_fileName)
+
+        # This is the path the the processed file created by Savu after having
+        # done the processing
+        savu_path = os.path.join(fileDir,
+                                 baseTemplate.replace('%s', "savuproc"))
```

Comment:
```suggestion
                                 os.path.splitext(fileTemplate)[0] % "savuproc")
```

Reply:
Do the file name templates ever contain %d (as per hdf name templates)?

Reply:
No, I never made that work, they only ever contain a single %s